src/ews.erl

```diff
@@ -142,21 +160,42 @@ record_to_map(Model, R) ->
 
 %% Add a pre-call hook which is called just before making the actual
 %% SOAP call. A pre hook is a function of one argument, which will be
-%% a list [EndPoint, Operation, EncodedHeader, EncodedBody, Options] where:
+%% a list [EndPoint, Operation, SoapAction, EncodedHeader, EncodedBody,
+%%         Options] where:
 %%  Endpoint:      The service endpoint that will be used for the call
 %%  Operation:     Service operation
-%%  EncodedHeader: Header after XML encoding
-%%  EncodedBody:   Body after XML encoding
+%%  SoapAction:    SOAPAction as string or binary, usually ""
+%%  EncodedHeader: Header after XML encoding in ews format
+%%  EncodedBody:   Body after XML encoding in ews format
 %%  Options:       Map supplied in the call_service_op call
 %% It should return a list of the same kind, with potentially updated
 %% values that are to be used in the call.
+%% The value returned for Operation will be ignored on return from hooks.
 %% Hooks are called in the order they were added, each hook being passed
 %% the output of the previous hook.
 add_pre_hook(Hook) ->
     ews_svc:add_pre_hook(default, Hook).
 add_pre_hook(Model, Hook) ->
     ews_svc:add_pre_hook(Model, Hook).
 
+%% Add a pre-call hook which is called just before making the actual
+%% SOAP call, and after Header and Body has been converted to XML in an iolist.
+%% A pre hook is a function of one argument, which will be
+%% a list [EndPoint, Operation, SOAP, Options] where:
```

Comment:
I don't like bike shedding naming, but I also really don't like the name "pre post hook".

The "pre hook" is called after encoding records to abstract XML documents but before the SOAP call (it operates on the request).

The "pre post hook" is called after the second encoding step of the soap request, when the XML is "binary" (or at least lost its abstract form and is now just text, but before the request is sent to the remote service.

The "post hook" is called after the SOAP call (it operates on the response), after the response has been decoded to abstract XML but before decoding to records.

So the "pre post hook" is more of a "post pre hook" :P Wouldn't `post_serialize_hook` or something be more clear?

Reply:
Yeah, I'm not super stoked about the name either. But I wanted it to start with `pre_` since it is a pre-hook that works on the SOAP before the call. The post hooks works after the call, so I didn't think it should start with `post_` at least.